File: app/src/test/java/com/werocksta/dagger2demo/JsonResource.java
package com.werocksta.dagger2demo;

import java.io.InputStream;

import okio.Okio;

public class JsonResource {
    @SuppressWarnings("NewApi")
    public static String fromResource(String file) throws Exception {
        ClassLoader classLoader = Thread.currentThread().getContextClassLoader();
        try (InputStream is = classLoader.getResourceAsStream(file)) {
            return Okio.buffer(Okio.source(is)).readUtf8();
        }
    }
}

File: app/src/test/java/com/werockstar/dagger2demo/JsonResource.kt
package com.werockstar.dagger2demo

import okio.Okio

object JsonResource {
    @Throws(Exception::class)
    fun fromResource(file: String): String {
        val classLoader = Thread.currentThread().contextClassLoader
        classLoader.getResourceAsStream(file).use { `is` -> return Okio.buffer(Okio.source(`is`)).readUtf8() }
    }
}
